### ISSUE TITLE
Using locally compiled provider instructions are not working

### ISSUE THREAD

**Issue by ahmetb** [hashicorp/terraform-provider-google#8052]
I'm reading https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#instructing-terraform-to-use-a-local-copy-of-the-provider and I am finding it to be really complex, also can't get it to work at all.

>  If you are building Terraform from source with a Google provider built from source, Terraform will automatically use the local terraform-provider-google and terraform-provider-google-beta plugins in $GOPATH/bin.

To verify this, I did this:

1. `git clone hashicorp/terraform`
2. `git install`
3. go to a working example
4. `rm -rf .terraform*`
5. clone this repo
6. `make`
7. verify binary is at `~/go/terraform-provider-google`

technically at this point, if I run `terraform init` I expect no `google` provider to be downloaded.

It's still downloaded...

```
- Finding hashicorp/google versions matching ">= 3.32.0, < 4.0.0"...
- Finding hashicorp/google-beta versions matching ">= 3.32.0, < 4.0.0"...
- Installing hashicorp/google v3.51.0...
- Installed hashicorp/google v3.51.0 (signed by HashiCorp)
- Installing hashicorp/google-beta v3.51.0...
- Installed hashicorp/google-beta v3.51.0 (signed by HashiCorp)
```

Also this section is quite hard to follow. there are bunch of dir names flying around like `~/.terraform.d` `$GOPATH/.terraform` `$GOPATH/bin/`... If someone can get to rewriting this section to be more prescriptive and concise, that would be appreciated by future contributors.

**Comment by rileykarson** [comment]
What Terraform version are you using? There's been a lot of churn in how local providers work recently, and those instructions will only work for `0.13`. `0.12` made it much simpler to use local providers, and I believe `0.14` aims to be easier than `0.13` but won't be as simple as `0.12`. Unfortunately, it gets there by breaking how they're set up for a second time in a row.

**Comment by ahmetb** [comment]
I’ve tried HEAD on Terraform repo (0.15 dev) as well as 0.14.
After spending a couple of hours, I mostly gave up.

**Comment by slevenick** [comment]
I was able to nearly exactly follow the steps on that page with Terraform version 1.1.3. I think there is a little environment-specific requirements based on where your locally built version of the provider ends up, but I don't think we can be more prescriptive about that. 

Personally I build my local version of the provider inside the terraform-provider-google(-beta) repos, so I changed the symlink to point there rather than using $GOPATH, but I believe it should work if a user's machine is set up with GOPATH. I don't think anything has changed within Terraform core to prevent the listed directions
